[Testing] Add ASDisplayNodeLayoutTests

maicki opened the pull request:
In preparation for a fix that is coming to `ASDisplayNode` layout mechanism let's start with having tests for the layout behavior of `ASDisplayNode`

cc @Adlai-Holler @garrettmoon

Adlai-Holler commented:
@maicki This is a great start! Yes!

Let's add this as a new header file, and standardize on those macros for geometry assertions: https://gist.github.com/Adlai-Holler/70f84195ca8d53b93c420e45dac0f498 That way we get good failure messages while debugging.

Also I'm not quite clear on what we're testing here. `testNodeLayoutDoesNotChangeIfNotMeasured`?

maicki commented:
@Adlai-Holler Addressed comments. In case it looks good, can you merge it? I would like to put a PR on top and the test(s) should be more clear in there.

Adlai-Holler commented:
Cool, merging!

Adlai-Holler commented on AsyncDisplayKitTests/ASXCTExtensions.h:
```diff
@@ -0,0 +1,37 @@
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+//
```
@maicki Let's yank this header stuff out, and move the stuff below up here. No need to be formal about landing it, feel free to just push to master

maicki replied:
@Adlai-Holler Can you take that over please. I don't have the access to upstream master on my Mac at the moment ... Thanks!

maicki replied:
@Adlai-Holler Here we go #2158 unfortunately I don't have access to direct upstream master at the moment
